### PR TITLE
Sincronize 2.8+ compatible changes made to 2.7x-support branch with master branch

### PR DESCRIPTION
**Pull request by sercero**
Synchronize recent changes made to the 2.7x-support branch:

- Support for add-on script reload
- Inhibit export of any user properties when EXPORT_USER is not selected
- Add Particle system random object distribution

### CODE REVIEW

**Review by paroj — CHANGES_REQUESTED** [review]

File: io_ogre/config.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
     'SEP_MATS' : True,
     'SELONLY' : True,
     'EXPORT_HIDDEN' : True,
+    'EXPORT_USER' : True,
```
[the remark this answers is not in the document]
Comment (sercero) [reply io_ogre/config.py:36]: There are other properties

File: io_ogre/ogre/scene.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def dot_scene(path, scene_name=None):
             # that only get spaces converted to _, just do that automatically.
             cleanname = clean_object_name(ob.name)
             cleannamespaces = clean_object_name(ob.name, spaces = False)
-
+            print("ABABA", ob.name)
```
[the remark this answers is not in the document]
Comment (sercero) [reply io_ogre/ogre/scene.py:56]: Do you know why that "ABABA" is there?

File: io_ogre/config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (paroj) [reply io_ogre/config.py:36]: let me rephrase: when exporting the cube with 2.80+ I get an empty `userData`. So to get extra properties, one has to explicitly request them and this additional toggle is not needed. In contrast, with 2.7x you always get some dummy physics values.

Getting rid of the empty `userData`, should be rather a check whether there are any properties to be exported.

[the remark this answers is not in the document]
Comment (sercero) [reply io_ogre/config.py:36]: OK, I understand

[the remark this answers is not in the document]
Comment (sercero) [reply io_ogre/config.py:36]: Pushed new changes
